### PR TITLE
Allow specifying the entityBaseURL.

### PR DESCRIPTION
**Pull request by holtgrewe**
Describe changes proposed in this pull request:
- allow specifying `entityBaseUrl` for `MetadataGenerator`; this appears to be required when deploying cBioPortal behind a reverse proxy that does the SSL handling.
- setting this to `#{null}` by default to keep the old behaviour

# Checks
- [ ] Runs on heroku
- [ ] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [ ] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [Giphy CAPTURE](https://giphy.com/apps/giphycapture) or [Peek](https://github.com/phw/peek)

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). It can help to figure out who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them either through slack or by assigning them as a reviewer on the PR

### CODE REVIEW

**Comment by holtgrewe** [comment]
@n1zea144 could you review this? You last edited the area around my change in `applicationProperties-security.xml`.

File: src/main/resources/portal.properties.EXAMPLE [hunk 1]
```diff
@@ -100,6 +100,7 @@ microsoftlive.consumer.secret=
 
 ## SAML settings
 saml.sp.metadata.entityid=
+saml.sp.metadata.entityBaseURL=#{null}
```
Comment (inodb) [inline src/main/resources/portal.properties.EXAMPLE:103]: @holtgrewe Thanks for the PR! Why should this one be `#{null}`? Is that the same as empty string? Like why should this value have to be different from the other saml example values?

Comment (inodb) [reply src/main/resources/portal.properties.EXAMPLE:103]: ah i see never mind, you put it in the description, for keeping the old behavior. Makes sense

Comment (inodb) [reply src/main/resources/portal.properties.EXAMPLE:103]: maybe good to put a comment in the example file to clarify? Something like
```suggestion
# change this url if behind reverse proxy that handles SSL, see docs/Authenticating-Users-via-SAML.md
saml.sp.metadata.entityBaseURL=#{null}
```

**Comment by holtgrewe** [comment]
@inodb thanks! Would it be possible to get a Docker container just for this patch? Somehow when I build them locally they do not startup properly (cannot connect to MySQL).

File: security/security-spring/src/main/resources/applicationContext-security.xml [hunk 2]
```diff
@@ -349,6 +349,7 @@
         <b:constructor-arg>
             <b:bean class="org.springframework.security.saml.metadata.MetadataGenerator">
                 <b:property name="entityId" value="${saml.sp.metadata.entityid}"/>
+                <b:property name="entityBaseURL" value="${saml.sp.metadata.entitybaseurl}"/>
```
Comment (inodb) [inline security/security-spring/src/main/resources/applicationContext-security.xml:352]: curious, would this btw work?
```suggestion
                <b:property name="entityBaseURL" value="${saml.sp.metadata.entitybaseurl:#{null}}"/>
```

**Comment by inodb** [comment]
@holtgrewe how are you building the container?

This should work from the root dir (best to clone the repo fresh):
```
 docker build . -t cbioportal/cbioportal:3.2.13 -f docker/web-and-data/Dockerfile
```

Make sure you don't have an existing `portal.properties` file before running it (mvn will use the default `portal.properties.EXAMPLE` in that case):

```
rm src/main/resources/portal.properties
```

You can also build directly with Docker (not 100% sure if this command works but this is approx what we use in our e2e tests):

```
docker build https://github.com/holtgrewe/cbioportal.git#ee865928db66a53a4e70a237c851a2a8b6a668b9 -f docker/web-and-data/Dockerfile -t holtgrewe/cbioportal:saml-entity-url
```

We release a new one every week on Thursday (so just missed this one). If that's too long you could also temporarily use the dev image `cbioportal/cbioportal:master` after this is merged.

**Pull request merged**